rules/Php81/Rector/Array_/ArrayToFirstClassCallableRector.php

```diff
@@ -174,4 +171,25 @@ private function isNonStaticOtherObject(
 
         return ! $extendedMethodReflection->isPublic();
     }
+
+    private function isReferenceToNonPublicMethodOutsideOwningScope(MethodCall $methodCall, string $methodName): bool
+    {
+        if ($methodCall->var instanceof Variable && $methodCall->var->name === 'this') {
```

Comment:
I've been intentionally restrictive here - the code will only be refactored if it explicitly references `$this`, I don't think it's sensible to try and check for e.g. other variables that have been defined as references to `$this`.

Reply:
ensure method exists, this can be any: `$name = [$this, 'whatever'];` which watever is never exists, which should be skipped as that means not a callable (normal array).